lib/http/check_factory.rb

```diff
@@ -1,66 +1,115 @@
-  end
+        #
+        # Provide the full list of checks
+        #
+        class << self
+          attr_reader :checks
+        end
 
-  #
-  # Provide the full list of checks
-  #
-  def self.generate_initial_checks(url)
-    @checks.map{ |x| x.new.generate_checks("#{url}") }.flatten.compact.select{|x|
-      x[:require_product] == nil && x[:require_vendor] == nil && x[:require_vendor_product] == nil }
-  end
+        #
+        # Provide the full list of checks
+        #
+        def self.generate_initial_checks(url, opts = {})
+          checks_to_return = @checks.map { |x| x.new.generate_checks(url.to_s) }.flatten.compact
 
-  #
-  # Provide checks given a vendor
-  #
-  def self.generate_checks_for_vendor(url,vendor)
-    @checks.map{ |x| x.new.generate_checks("#{url}") }.flatten.compact.select{|x|
-      x[:require_vendor] == "#{vendor}" }
-  end
+          # if tags were provided, filter by them
+          checks_to_return = filter_by_tags(checks_to_return, opts)
 
-  #
-  # Provide checks givene a product
-  #
-  def self.generate_checks_for_product(url,product)
-    @checks.map{ |x| x.new.generate_checks("#{url}") }.flatten.compact.select{|x|
-      x[:require_product] == "#{product}" }
-  end
+          # if we're not in noisy mode, filter checks that have require_* conditions
+          unless opts[:noisy]
+            checks_to_return = checks_to_return.select do |x|
+              x[:require_product].nil? && x[:require_vendor].nil? && x[:require_vendor_product].nil? && x
+            end
+          end
 
+          checks_to_return
+        end
 
-  #
-  # Provide checks givene a vendor product
-  #
-  def self.generate_checks_for_vendor_product(url, vendor, product)
-    @checks.map{ |x| x.new.generate_checks("#{url}") }.flatten.compact.select{|x|
-      x[:require_vendor_product] == "#{vendor}_#{product}".downcase.gsub(" ","_") }
-  end
+        #
+        # Provide checks given a vendor
+        #
+        def self.generate_checks_for_vendor(url, vendor, opts = {})
+          checks_to_return = @checks.map { |x| x.new.generate_checks(url.to_s) }.flatten.compact
 
+          # if tags were provided, filter by them
+          checks_to_return = filter_by_tags(checks_to_return, opts)
 
-  #
-  # Provide the full list of checks
-  #
-  def self.configuration_checks
-    @checks.map{ |x| x.new.generate_checks("") }.flatten.compact.select{|x| x[:type] == "content" }
-  end
+          # if we're not in noisy mode, filter checks that have require_* conditions
+          unless opts[:noisy]
+            checks_to_return = checks_to_return.select do |x|
+              x[:require_vendor] == vendor.to_s
+            end
+          end
 
+          checks_to_return
+        end
 
-end
-end
-end
+        #
+        # Provide checks givene a product
+        #
+        def self.generate_checks_for_product(url, product, opts = {})
+          checks_to_return = @checks.map { |x| x.new.generate_checks(url.to_s) }.flatten.compact
```

Comment:
for the functions that generate checks based on vendor/product, we agreed that we're ignoring noisy mode. Please adjust.